app/controllers/carts_controller.rb

```diff
@@ -1,25 +1,27 @@
 class CartsController < ApplicationController
   def create
-    if session[:cart]
-      if session[:cart][params[:item_id]]
-        session[:cart][params[:item_id]]  += 1
-      else
-        session[:cart][params[:item_id]] = 1
-      end
-    else
-      session[:cart] = {params[:item_id] => 1}
-    end
+    session[:cart] ||= {}
+    session[:cart][params[:item_id]] ||= 0
+    session[:cart][params[:item_id]] += 1
     @cart_items = session[:cart]
 
-    flash[:notice] = 'Added '  + Item.find_by_dao(params[:item_id]).name
+    flash[:notice] = 'Added ' + Item.find_by_dao(params[:item_id]).name
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.